### PR TITLE
[QOL] Have Friendship Value be shown in summary

### PR DESCRIPTION
**Pull request by CodeTappert**
## What are the changes the user will see?

The current friendship value of a pokemon in the summary

## Why am I making these changes?

Because it is kinda annoying to not know how far you pokemon currently is till its evo

## What are the changes from a developer perspective?

I copied the code from candy icon and while talking to damo placed it where it is.

### Screenshots/Videos

image

## How to test the changes?

Open the summary

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [ ] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by CodeTappert** [comment]
Draft till we have the icon from the art team

File: src/locales/de/pokemon-summary.json [hunk 1]
```diff
@@ -40,5 +40,6 @@
     "Sassy": "{{nature}}",
     "Careful": "{{nature}}",
     "Quirky": "{{nature}}"
-  }
+  },
+  "friendship": "Freundschaft"
```
Comment (CodeTappert) [inline src/locales/de/pokemon-summary.json:44]:
```suggestion
  }
```

Comment (CodeTappert) [inline src/locales/de/pokemon-summary.json:44]:
```suggestion
  }
```

**Comment by CodeTappert** [comment]
No real issues with this to discuss. It could only be problematic with super big pokemon. But thats the same for the candy counter so...

File: src/ui/summary-ui-handler.ts [hunk 2]
```diff
@@ -345,6 +373,15 @@ export default class SummaryUiHandler extends UiHandler {
 
     this.candyShadow.setCrop(0, 0, 16, candyCropY);
 
+    if (this.friendshipShadow.visible) {
+      this.friendshipShadow.on("pointerover", () => (this.scene as BattleScene).ui.showTooltip("", `${i18next.t("pokemonSummary:friendship")}`, true));
+      this.friendshipShadow.on("pointerout", () => (this.scene as BattleScene).ui.hideTooltip());
+    }
+
+    this.friendshipText.setText(` ${this.pokemon?.friendship||"0"}/255`);
```
Comment (torranx) [inline src/ui/summary-ui-handler.ts:381]: nit: formatting ( I should really update my pr :pokedollAh: )
```suggestion
    this.friendshipText.setText(`${this.pokemon?.friendship || "0"} / 255`);
```

File: src/ui/summary-ui-handler.ts [hunk 3]
```diff
@@ -345,6 +373,15 @@ export default class SummaryUiHandler extends UiHandler {
...
+
+    this.friendshipShadow.setCrop(0, 0, 16, 16 - (16 * ((this.pokemon?.friendship||0) / 255)));
```
Comment (torranx) [inline src/ui/summary-ui-handler.ts:383]: nit
```suggestion
    this.friendshipShadow.setCrop(0, 0, 16, 16 - (16 * ((this.pokemon?.friendship || 0) / 255)));
```

File: src/ui/summary-ui-handler.ts [hunk 4]
```diff
@@ -292,6 +317,9 @@ export default class SummaryUiHandler extends UiHandler {
     this.candyIcon.setTint(argbFromRgba(Utils.rgbHexToRgba(colorScheme[0])));
     this.candyOverlay.setTint(argbFromRgba(Utils.rgbHexToRgba(colorScheme[1])));
 
+    this.friendshipIcon.setTint(argbFromRgba(Utils.rgbHexToRgba(colorScheme[0])));
```
Comment (MokaStitcher) [inline src/ui/summary-ui-handler.ts:320]: Should we really tint the icon like this? It makes it hard to see how full the heart actually is for some mons, like this Garchomp
Screenshot 2024-09-30 at 20 29 19 Screenshot 2024-09-30 at 20 29 25

File: src/ui/summary-ui-handler.ts [hunk 5]
```diff
@@ -345,6 +373,15 @@ export default class SummaryUiHandler extends UiHandler {
 
     this.candyShadow.setCrop(0, 0, 16, candyCropY);
 
+    if (this.friendshipShadow.visible) {
+      this.friendshipShadow.on("pointerover", () => (this.scene as BattleScene).ui.showTooltip("", `${i18next.t("pokemonSummary:friendship")}`, true));
+      this.friendshipShadow.on("pointerout", () => (this.scene as BattleScene).ui.hideTooltip());
```
Comment (MokaStitcher) [inline src/ui/summary-ui-handler.ts:378]:
```suggestion
      this.friendshipShadow.on("pointerover", () => this.scene.ui.showTooltip("", `${i18next.t("pokemonSummary:friendship")}`, true));
      this.friendshipShadow.on("pointerout", () => this.scene.ui.hideTooltip());
```

Pretty sure this.scene is a BattleScene in UiHandlers. Just double check if possible, and might as well update it for the candy tooltip as well if it's the case

File: src/ui/summary-ui-handler.ts [hunk 6]
```diff
@@ -187,6 +191,27 @@ export default class SummaryUiHandler extends UiHandler {
...
+    this.friendshipShadow.setScale(0.8);
+    this.friendshipShadow.setInteractive(new Phaser.Geom.Rectangle(0, 0, 16, 16), Phaser.Geom.Rectangle.Contains);
+    this.summaryContainer.add(this.friendshipShadow);
+
+    this.friendshipText = addTextObject(this.scene, 20, -66, "x0", TextStyle.WINDOW_ALT, { fontSize: "76px" });
+    this.friendshipText.setOrigin(0, 0);
+    this.summaryContainer.add(this.friendshipText);
+
+
+
```
Comment (MokaStitcher) [inline src/ui/summary-ui-handler.ts:214]:
```suggestion
```

bit too many empty lines here :p

**Review by MokaStitcher — CHANGES_REQUESTED** [review]
Just some doubts about whether we should tint the icon + a few nits

This is already the case for the candy tooltip, but might as well fix it while we're at it:
If you leave the summary with a tooltip open it doesn't get cleared. Should be easily fixable by calling `ui.hideTooltip()` in the handler's `clear()` function (I think. If not maybe don't worry about it)

https://github.com/user-attachments/assets/0792fb80-290c-43c4-ad03-425d3d1c13e5

File: src/ui/summary-ui-handler.ts (hunk 4 above)
Comment (CodeTappert) [reply src/ui/summary-ui-handler.ts:320]: No we should not! I just copied the code from candy and wanted remove that. Thank you

Comment (CodeTappert) [inline src/ui/summary-ui-handler.ts:320]:
```suggestion
```

**Review by torranx — APPROVED** [review]

**Review by MokaStitcher — APPROVED** [review]

**Comment by MokaStitcher** [comment]
For Temp, still some doubts about the friendship location [here](https://discord.com/channels/1125469663833370665/1289575708066058374/1290382247891046452), not the one to make the call of whether this is fine for now or not

**Comment by DayKev** [comment]
Can you replace the image in the OP with an updated one?

File: src/locales/en/pokemon-summary.json [hunk 7]
```diff
@@ -40,5 +40,6 @@
     "Sassy": "{{nature}}",
     "Careful": "{{nature}}",
     "Quirky": "{{nature}}"
-  }
+  },
+  "friendship": "Friendship"
```
Comment (CodeTappert) [inline src/locales/en/pokemon-summary.json:44]:
```suggestion
  }
```

**Comment by CodeTappert** [comment]
> Just some doubts about whether we should tint the icon + a few nits
> 
> This is already the case for the candy tooltip, but might as well fix it while we're at it: If you leave the summary with a tooltip open it doesn't get cleared. Should be easily fixable by calling `ui.hideTooltip()` in the handler's `clear()` function (I think. If not maybe don't worry about it)
> 
>  tooltip_clear.mov

The text staying is a problem with our hovers at all. Not with this specific one

**Review by flx-sta — APPROVED** [review]

**Comment by flx-sta** [comment]
It begs the question how we going to update the sub-module reference to include the newly added keys

**Review by DayKev — APPROVED** [review]

**Pull request merged**